src/Middleware/ValidateWebAppData.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Nutgram\Laravel\Middleware;
+
+use Closure;
+use Illuminate\Http\Request;
+use SergiX44\Nutgram\Exception\InvalidDataException;
+use SergiX44\Nutgram\Nutgram;
+
+class ValidateWebAppData
+{
+    public function handle(Request $request, Closure $next)
+    {
+        try {
+            $initData = $request->input('initData', '');
+            $data = app(Nutgram::class)->validateWebAppData($initData);
+
+            $request->attributes->add(['webapp' => $data]);
+            return $next($request);
+        } catch (InvalidDataException) {
+            abort(403);
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏ would put that inside separated method, so anyone is free to extend the middleware and define the behaviour in case of failure